### PR TITLE
Add webp support

### PR DESCRIPTION
**Pull request by dualface**
> WebP is a new image format that provides lossless and lossy compression for images on the web. WebP lossless images are 28% smaller in size compared to PNGs. WebP lossy images are 25-34% smaller in size compared to JPEG images at equivalent SSIM index. WebP supports lossless transparency (also known as alpha channel) with just 22% additional bytes. WebP also supports features such as animation, ICC color profile, XMP meta-data and tiling.
> 
> https://developers.google.com/speed/standards/
### ENCODE IMAGE TO WebP

```
cwebp hello.png -o hello.webp
```
### LOAD WebP IN COCOS2D-X

```
CCSprite *sprite = CCSprite::create("hello.webp");
CCTextureCache::sharedTextureCache()->addImage("sheet.webp");
```

FOR RGB565, PLEASE USE PNG.

THANKS:

KunLun, Hui Wang

### CODE REVIEW

**Comment by martell** [comment]
Looks good.
I done this a few months ago but only for the windows platform

**Comment by dualface** [comment]
tests on ios/android/mac/windows ok :-)

File: cocos2dx/platform/ios/CCImage.mm [hunk 1]
```diff
@@ -264,7 +264,13 @@ static bool _initWithString(const char * pText, cocos2d::CCImage::ETextAlign eAl
...
         {
             dim.height = constrainSize.height;
-        }         
+        }
+        
+        dim.width = (int)(dim.width / 2) * 2 + 2;
+        dim.height = (int)(dim.height / 2) * 2 + 2;
+        
+        dim.width = (int)(dim.width / 2) * 2 + 2;
+        dim.height = (int)(dim.height / 2) * 2 + 2;
```
Comment (minggo) [inline cocos2dx/platform/ios/CCImage.mm]: Duplicate codes?

**Comment by dualface** [comment]
Sorry, my mistake. I will fix ASAP.

**Comment by dualface** [comment]
Duplicate codes removed :)

**Pull request merged**